includes/shallowbackup.js

```diff
@@ -14,61 +14,78 @@
 'use strict';
 
 const async = require('async');
+const error = require('./error.js');
+const events = require('events');
 const request = require('./request.js');
 
-module.exports = function(dbUrl, blocksize, parallelism, log, resume) {
+module.exports = function(dbUrl, limit, parallelism, log, resume) {
+  if (typeof limit === 'string') limit = parseInt(limit);
+
   const client = request.client(dbUrl, parallelism);
-  if (typeof blocksize === 'string') {
-    blocksize = parseInt(blocksize);
-  }
-  const events = require('events');
   const ee = new events.EventEmitter();
-
-  var startdocid = null;
   const start = new Date().getTime();
-  var batch = 1;
+  var batch = 0;
+  var hasErrored = false;
+  var startKey = null;
   var total = 0;
 
-  async.doUntil(function(callback) {
-    var opts = {limit: blocksize + 1, include_docs: true};
-    if (startdocid) {
-      opts.startkey_docid = startdocid;
-    }
-    var r = {
-      url: dbUrl + '/_all_docs',
-      method: 'GET',
-      qs: opts
-    };
-    client(r, function(err, res, data) {
-      if (err || !data.rows) {
-        ee.emit('error', err || data);
-        return callback(null, null);
-      }
+  async.doUntil(
+    function(callback) {
+      var opts = {limit: limit, include_docs: true};
+      if (startKey) opts.startkey = `"${startKey}\\u0000"`; // exclusive start key
```

Comment:
This could probably use a comment, as it is a bit of trick.
> To avoid double fetching a document solely for the purposes of getting the next ID to use as a `startkey` for the next page we instead use the last ID of the current page and append the lowest unicode sort character.